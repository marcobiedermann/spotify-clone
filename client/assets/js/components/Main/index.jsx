import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.css';

const Main = ({ children }) => (
  <main className={styles.main}>
    {children}
  </main>
);

Main.propTypes = {
  children: PropTypes.node,
};

Main.defaultProps = {
  children: null,
};

export default Main;
