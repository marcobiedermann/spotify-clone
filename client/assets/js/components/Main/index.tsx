import classNames from 'classnames';
import React, { FC } from 'react';
import styles from './style.css';

export interface MainProps {
  className?: string;
}

const Main: FC<MainProps> = (props) => {
  const { className, ...otherProps } = props;

  return <main className={classNames(className, styles.main)} {...otherProps} />;
};

export default Main;
