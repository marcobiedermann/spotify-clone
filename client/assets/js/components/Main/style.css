.main {
  flex-grow: 1;
  padding-left: 1.5em;
  padding-right: 1.5em;
}
