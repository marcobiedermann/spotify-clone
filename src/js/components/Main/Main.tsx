import clsx from 'clsx';
import React, { ReactNode } from 'react';
import styles from './Main.module.css';

export interface MainProps {
  children: ReactNode;
  className?: string;
}

function Main(props: MainProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <main className={clsx(className, styles.main)} {...otherProps} />;
}

export default Main;
