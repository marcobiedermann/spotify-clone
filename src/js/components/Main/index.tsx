import classNames from 'classnames';
import React, { ReactNode } from 'react';
import styles from './style.module.css';

export interface MainProps {
  children: ReactNode;
  className?: string;
}

function Main(props: MainProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <main className={classNames(className, styles.main)} {...otherProps} />;
}

export default Main;
