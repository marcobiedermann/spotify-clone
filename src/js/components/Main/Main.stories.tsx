import type { Meta, StoryObj } from '@storybook/react';
import Main from '.';

const meta: Meta<typeof Main> = {
  component: Main,
  args: {
    children: 'Main',
  },
};

type Story = StoryObj<typeof Main>;

const Default: Story = {};

export { Default };
export default meta;
