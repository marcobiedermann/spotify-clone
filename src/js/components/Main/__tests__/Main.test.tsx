import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';
import Main from '..';

it('renders correctly', () => {
  const { container } = render(<Main>Main</Main>);

  expect(container.firstChild).toMatchSnapshot();
});
