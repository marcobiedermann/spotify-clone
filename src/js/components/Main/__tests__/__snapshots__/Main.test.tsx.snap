// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<main
  class="main"
>
  Main
</main>
`;
